src/ingest_validation_tests/segmentation_mask_imagesize_validation.py

```diff
@@ -0,0 +1,90 @@
+from pathlib import Path
+from typing import List, Optional
+
+import tifffile
+import xmlschema
+from ingest_validation_tools.plugin_validator import Validator
+from utils import GetParentData
+
+
+def get_ometiff_size(file) -> Optional[str, dict]:
+    try:
+        tf = tifffile.TiffFile(file)
+        xml_document = xmlschema.XmlDocument(tf.ome_metadata)
+        if xml_document.schema and not xml_document.schema.is_valid(xml_document):
+            return f"{file} is not a valid OME.TIFF file"
+    except Exception as excp:
+        return f"{file} is not a valid OME.TIFF file: {excp}"
+    xml_image_data = xml_document.schema.to_dict(xml_document).get("Image")[0].get("Pixels")
+    try:
+        rst = {
+            "X": xml_image_data.get("@PhysicalSizeX"),
+            "XUnits": xml_image_data.get("@PhysicalSizeXUnits"),
+            "Y": xml_image_data.get("@PhysicalSizeY"),
+            "YUnits": xml_image_data.get("@PhysicalSizeYUnits"),
+            "Z": xml_image_data.get("@PhysicalSizeZ"),
+            "ZUnits": xml_image_data.get("@PhysicalSizeZUnits"),
+        }
+        return rst
+    except Exception as excp:
+        return f"{file} is not a valid OME.TIFF file: {excp}"
+
+
+class ImageSizeValidator(Validator):
+    description = "Check dataset and parent image size so they can be matched in the visualization"
+    cost = 1.0
+    version = "1.0"
+    required = "segmentation_mask"
+
+    def collect_errors(self, **kwargs) -> List[Optional[str]]:
+        del kwargs
+        if self.required not in self.contains and self.assay_type.lower() != self.required:
+            return []  # We only test Segmentation Masks
+        files_tested = None
+        output = []
+        filenames_to_test = []
+        parent_filenames_to_test = []
+        try:
+            for row in self.metadata_tsv.rows:
+                data_path = Path(row["data_path"])
+                if not data_path.is_absolute():
+                    data_path = Path(self.paths[0]).parent / data_path
+
+                for glob_expr in [
```

Comment:
Maybe consolidate this list into a class attribute or something unless there's a chance we'd want to have different patterns for mask/base images.

Reply:
I've refactored this a bit, how does it look now?